collectors/o365/o365_collector.js

```diff
@@ -164,6 +164,15 @@ class O365Collector extends PawsCollector {
 
             return callback(null, logs, newState, newState.poll_interval_sec);
         }).catch(err => {
+            if (err.message) {
+                let message = err.message;
+                try {
+                    let error = JSON.parse(message.slice(message.indexOf('{'), message.lastIndexOf('}') + 1));
+                    err.errorCode = error.error;
+                } catch (exception) {
```

Comment:
I wouldnt return the parsing exception because that will mask the actual collector error.

This try/catch block is more for handling parsing errors gracefully. I would return the `err` object in the callback.

Reply:
okay i will update.